Add aten::to NNAPI converter

axitkhurana opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61025
* #61024
* #61023
* #61022
* #61021
* #59529
* #59434
* #59365
* #59364
* #58543
* #60885
* #58541
* **#58540**
* #58539

Summary:

Add support for aten::to op in the NNAPI model converter for simple
cases like to("cpu"), to("gpu")

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_to

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

Commit:
aten::to

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531941).

Commit:
Update on "aten::to"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

Commit:
Update on "aten::to"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531941).

Commit:
Update on "aten::to"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

Commit:
Update on "aten::to"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531941).

Commit:
Update on "Add aten::to NNAPI converter"


Summary:

Add support for aten::to op in the NNAPI model converter for simple
cases like to("cpu"), to("gpu")

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_to

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

Commit:
Update on "Add aten::to NNAPI converter"


Summary:

Add support for aten::to op in the NNAPI model converter for simple
cases like to("cpu"), to("gpu")

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_to

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531941).

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531941).

Commit:
Update on "Add aten::to NNAPI converter"


Summary:

Add support for aten::to op in the NNAPI model converter for simple
cases like to("cpu"), to("gpu")

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_to

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531941).

Commit:
Update on "Add aten::to NNAPI converter"


Summary:

Add support for aten::to op in the NNAPI model converter for simple
cases like to("cpu"), to("gpu")

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_to

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531941).

Commit:
Update on "Add aten::to NNAPI converter"


Summary:

Add support for aten::to op in the NNAPI model converter for simple
cases like to("cpu"), to("gpu")

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_to

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531941).

dreiss commented on torch/backends/_nnapi/serializer.py:
```diff
@@ -877,6 +884,10 @@ def add_unsqueeze(self, node):
 
         self.add_operation(NNAPI_OperationCode.EXPAND_DIMS, inputs, outputs)
 
+    def add_to(self, node):
```
Maybe check that it's not trying to change dtype?

Commit:
Update on "Add aten::to NNAPI converter"


Summary:

Add support for aten::to op in the NNAPI model converter for simple
cases like to("cpu"), to("gpu")

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_to

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

Commit:
Update on "Add aten::to NNAPI converter"


Summary:

Add support for aten::to op in the NNAPI model converter for simple
cases like to("cpu"), to("gpu")

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_to

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

Commit:
Update on "Add aten::to NNAPI converter"


Summary:

Add support for aten::to op in the NNAPI model converter for simple
cases like to("cpu"), to("gpu")

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_to

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531941](https://our.internmc.facebook.com/intern/diff/D28531941)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531941).
